backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/TorClient.kt

```diff
@@ -36,12 +38,12 @@ class TorClient private constructor(
      * Calling this method is usually preferable to creating a completely separate
      * `TorClient` instance, since it can share its internals with the existing `TorClient`.
      */
-    suspend fun isolatedClient(): TorClient =
-        accessMutex.withLock {
-            withContext(Dispatchers.IO) {
-                checkNotNull(nativeHandle) { "TorClient is disposed" }
-                TorClient(nativeHandle = isolatedClient(nativeHandle!!))
-            }
+    suspend fun isolatedClient(): TorClient = accessMutex.withLock { isolatedClientInternal() }
+
+    private suspend fun isolatedClientInternal() =
```

Comment:
Document this to specify that the caller MUST acquire `accessMutex` before calling this method. (Yes it's a private method, but this helps us in the future to not forget.)